test/unit_test/components/servo_test.dart

```diff
@@ -0,0 +1,228 @@
+import 'package:viam_sdk/src/components/servo/service.dart';
+import 'package:viam_sdk/src/components/servo/servo.dart';
+import 'package:viam_sdk/src/gen/component/servo/v1/servo.pbgrpc.dart';
+import 'package:viam_sdk/src/resource/manager.dart';
+
+class FakeServo extends Fake implements Servo {
+  int angle = 0;
+  bool isStopped = true;
+  Map<String, dynamic>? extra;
+
+  @override
+  final String name;
+
+  FakeServo(this.name);
+
+  @override
+  Future<void> move(int angle, {Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    this.angle = angle;
+    isStopped = false;
+  }
+
+  @override
+  Future<int> position({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return angle;
+  }
+
+  @override
+  Future<bool> isMoving() async {
+    return !isStopped;
+  }
+
+  @override
+  Future<void> stop({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    isStopped = true;
+  }
+}
+
+void main() {
+  group('FakeServo Tests', () {
+    late FakeServo servo;
+    late String name;
+    const int testPosition = 42;
+    setUp(() {
+      name = 'servo';
+      servo = FakeServo(name);
+    });
+    test('angle should start at 0', () {
+      expect(servo.angle, 0);
+    });
+
+    test('isStopped should start at true', () {
+      expect(servo.isStopped, true);
+    });
+
+    test('name should be constructed with the value passed in', () {
+      servo = FakeServo('test');
+      expect(servo.name, 'test');
+    });
+
+    test('move should move angle to new position', () async {
+      await servo.move(testPosition);
+      expect(servo.angle, testPosition);
+    });
+
+    test('move should set isStopped to false', () async {
+      await servo.move(testPosition);
+      expect(servo.isStopped, false);
+    });
+
+    test('stop should set isStopped to true', () async {
+      servo.isStopped = false;
+      await servo.stop();
+      expect(servo.isStopped, true);
+    });
+
+    test('getPosition should return the angle', () async {
+      int testResult = await servo.position();
+      expect(testResult, servo.angle);
+    });
+
+    test('doCommand', () async {
+      // TODO implement servo doCommand
+      expect(true, true);
+    });
+  });
+
+  group('Servo Service Tests', () {
+    late ClientChannel channel;
+    late FakeServo servo;
+    late ServoService service;
+    late Server server;
+    late String name;
+    const int testPosition = 42;
+
+    setUp(() async {
+      name = 'servo';
+      servo = FakeServo(name);
+      ResourceManager manager = ResourceManager();
+      manager.register(Servo.getResourceName(name), servo);
+
+      service = ServoService(manager);
+
+      channel = ClientChannel(
+        'localhost',
+        port: 50051,
+        options: ChannelOptions(
+          credentials: const ChannelCredentials.insecure(),
+          codecRegistry: CodecRegistry(codecs: const [GzipCodec(), IdentityCodec()]),
+        ),
+      );
```

Comment:
might be good to refactor this bit of test utility code out

Reply:
Good call out, thank you! I added a group wrapping these two groups and abstracted it out to there.